webapp/src/components/viewHeader/viewHeaderGroupByMenu.tsx

```diff
@@ -21,6 +26,26 @@ type Props = {
 const ViewHeaderGroupByMenu = (props: Props) => {
     const {properties, activeView, groupByProperty} = props
     const intl = useIntl()
+
+    const cards = useAppSelector(getCurrentViewCardsSortedFilteredAndGrouped)
+    const {visible: visibleGroups, hidden: hiddenGroups} = getVisibleAndHiddenGroups(cards, activeView.fields.visibleOptionIds, activeView.fields.hiddenOptionIds, groupByProperty)
+
+    const emptyVisibleGroups = visibleGroups.filter((g) => !g.cards.length)
+    const emptyVisibleGroupsCount = emptyVisibleGroups.length
+    const hiddenGroupsCount = hiddenGroups.length
+
+    const handleToggleGroups = (show: boolean) => {
```

Comment:
nit: Because this is not memoized this is forcing the re-render of the Menu.Text components. Not a big deal, but maybe an improvement later. (Actually not only is not memoized also is you are using them inside an anonymous function, so you should address that too).

Reply:
Anyway, you are following the same patterns showing in the rest of this component, so probably the best option is to leave it as is now, and in the future address it if it's needed.